app/src/components/submissionPage/SubmissionTextBox.test.tsx

```diff
@@ -0,0 +1,180 @@
+    container = document.createElement("div");
+    document.body.appendChild(container);
+});
+
+afterEach(() => {
+    // cleanup on exiting
+    unmountComponentAtNode(container);
+    container.remove();
+    container = null;
+});
+
+it("renders without crashing", () => {
+    act(() => {
+        render(
+            <SubmissionTextBox
+                snappables={[{ id: "test-id", display: "test-display" }]}
+                user={"test-user"}
+            />,
+            container
+        );
+    });
+    expect(container.children.length).toBeGreaterThan(0);
+});
+
+describe("CharactersLeftDisplay", () => {
+    test("characters remaining string renders", () => {
+        const elem1: MentionElements = { id: "idlength10", display: "Jim" };
+        const elem2: MentionElements = {
+            id: "anna@softwire.com",
+            display: "Anna",
+        };
+        const msg2: String =
+            "Hi there @[Anna](anna@softwire.com) and @[Jim](idlength10), you have cool voices!";
+        const component = renderer.create(
+            <CharactersLeftDisplay
+                snappedUsers={[elem1, elem2]}
+                message={msg2}
+            />
+        );
+        let tree = component.toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+    test("characters remaining string renders", () => {
+        const elem1: MentionElements = { id: "idlength10", display: "Jim" };
+        const elem2: MentionElements = {
+            id: "anna@softwire.com",
+            display: "Anna",
+        };
+        const msg1: String = "";
+        const component = renderer.create(
+            <CharactersLeftDisplay snappedUsers={[]} message={msg1} />
+        );
+        let tree = component.toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+});
+
+describe("OnSubmitMessageDisplay", () => {
+    test("Submit message appears if confirmation is true", () => {
+        librender(<OnSubmitMessageDisplay confirmation={true} />);
+        screen.getByText("Snap submitted!");
+    });
+    test("Submit message is empty if confirmation is false", () => {
+        librender(<OnSubmitMessageDisplay confirmation={false} />);
+        const text = screen.queryByText("Snap submitted!");
+        expect(text).toBe(null);
+    });
+});
+
+describe("SubmissionBoxErrorDisplay", () => {
+    test("test for appearance of error message on error", () => {
+        librender(<SubmissionBoxErrorDisplay error={"error"} />);
+        screen.getByText("error");
+        const text = screen.queryAllByText(/./);
+        expect(text.length).toBeGreaterThan(0);
+    });
+    test("test for no error message if no error", () => {
+        librender(<SubmissionBoxErrorDisplay error={""} />);
+        const text = screen.queryAllByText(/./);
+        expect(text.length).toBe(0);
+    });
+});
+
+/*  */
+describe("<SubmissionTextBox />", () => {
+    test("test for successful submission of a legitimate snap", () => {
```

Comment:
This isn't actually testing the `<SubmissionTextBox />` component is it?

In fact, all three of this tests appear to be testing validation rather than the text box component and should probably be in a separate test file.